src/main/xar-resources/data/lucene/lucene.xml

```diff
@@ -394,14 +394,30 @@
                     for the expression is set to this element.</para>
                 <para>The expression is evaluated and for each result item, a facet value is added
-                    to the dimension using the string value of the item. If the expression returns
-                    the empty sequence  for the current parent node, the corresponding facet will be
-                    empty as well.</para>
+                    to the dimension using the string value of the item. Therefore if the expression
+                    returns multiple items, a facet for that dimension will also hold multiple values.
+                    If the expression returns the empty sequence  for the current parent node, the corresponding
+                    facet will be empty as well.</para>
                 <para>A facet can also be defined to be hierarchical. A typical example would be a date, which consists of a year, month and day component. By
                     indexing the single components as separate parts of a hierarchical facet, we enable the user to drill down by year first, then by month and
                     finally by day. Let's assume each of our docbook articles has a <tag>pubdate</tag> containing a date in <code>xs:date</code>
                     format:</para>
                 <programlisting language="xml" xlink:href="listings/listing-52.txt"></programlisting>
+                <para>Hierarchical facets may also hold multiple values, for example if we would like to associate
+                    our documents with a subject classification on various levels of granularity (say: <emphasis>science</emphasis> with
+                    <emphasis>math</emphasis> and <emphasis>physics</emphasis> as subcategories or <emphasis>humanities</emphasis> with
+                    <emphasis>art</emphasis>, <emphasis>sociology</emphasis> and <emphasis>history</emphasis>).
+                    This way we enable the user to drill down into broad <emphasis>humanities</emphasis>
+                    or <emphasis>science</emphasis> subject first and choose particular topics afterwards.
+                    If the result of the hierarchical facet <code>expression</code>
+                    evaluates to an array, each of array members will be treated as a hierarchical value for that facet.
+                    Such an array could look in XQuery similar to <code>[('science', 'math'), ('humanities', 'history')]</code> and be
+                    a result of evaluationg a function like <code>idx:subject-hierarchy</code> below stored in an imported module (see <link linkend="external-module">below</link>)
+                    </para>
+                <programlisting language="xml" xlink:href="listings/listing-520.xml"/>
+                <programlisting language="xquery" xlink:href="listings/listing-521.txt"/>
+                <para>which assumes hierarchical subject structure stored in <emphasis>/db/lucenetest/subjects.xml</emphasis></para>
```

Comment:
Correction: I think the path referred to in listing-521.txt is `/db/subjects/subjects.xml`